finder-app/finder.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+#DIRECTORY - ARGUMENT 1
+filesdir=$1
+
+#STRING TO MATCH - ARGUMENT 2
+searchstr=$2
+
+# VARIABLE THAT INCREMENTS WHEN A MATCH IS FOUND IN A PARTICULAR FILE
+totalfiles=0
+
+#VARIABLE TO CAPTURE TOTAL LINES CONTAINING SEARCHSTRING
+linesmatch=0
+
+#TEMPORARY VARIABLE THAT HOLDS MATCHING LINES COUNT FOR A SINGLE FILE
+match=0
+
+
+##################### ERROR CHECKS ########################
+if [ $# -lt 2 ] || [ $# -gt 2 ]
+then
+   # echo "Argument mismatch. Give the arguments as follows:"
+    #echo "arg1 : directory path          arg2:search string parameter"
+    exit 1
+fi
+
+if [ ! -d $filesdir ]
+then
+        #echo"directory found"
+#else
+        #echo "directory NOT found"
+        exit 1
+fi
+#############################################################
+
+
+
+############## FUNCTION THAT IS CALLED RECURSIVELY TO TRAVERSE THROUGH SUBFOLDERS AND SEARCH STRING ##############
+function loop_and_check_string() {
```

Comment:
@Sharath-Jonnala consider whether you could simplify the logic here with some arguments to grep instead of your own logic.